package gtkclient;

message ICMS {
	required double ts = 1; // timestamp
	required uint64 tick = 2; // tdt tick
	required uint32 stim_chan = 3; // stimulation channel (1-indexed)
	repeated ICMS_artifact artifact = 4; // artifact waveform
}

message ICMS_artifact {
	required uint32 rec_chan = 1; // recording channel (1-indexed)
	repeated float sample = 2 [packed=true]; // artifact waveform sample
}
